workos/event_objects/directory_user.py

```diff
@@ -0,0 +1,77 @@
+from typing import Optional, List
+from enum import Enum
+from workos.utils.types import JsonDict
+
+
+class DirectoryUserRole:
+    def __init__(self, attributes: JsonDict) -> None:
+        self.slug: str = attributes["slug"]
+
+
+class DirectoryUserEmail:
+    def __init__(self, attributes: JsonDict) -> None:
+        self.type: Optional[str] = attributes.get("type")
+        self.value: Optional[str] = attributes.get("value")
+        self.primary: Optional[bool] = attributes.get("primary")
+
+
+class DirectoryUserState(Enum):
+    ACTIVE = "active"
+    INACTIVE = "inactive"
+
+
+class DirectoryUserEvent:
+    def __init__(self, attributes) -> None:
+        self.id: str = attributes["id"]
+        self.name: str = attributes["name"]
+        self.idp_id: str = attributes["idp_id"]
+        self.directory_id: str = attributes["directory_id"]
+        self.organization_id: str = attributes["organization_id"]
+        self.first_name: Optional[str] = attributes.get("first_name")
+        self.last_name: Optional[str] = attributes.get("last_name")
+        self.job_title: Optional[str] = attributes.get("job_title")
+        self.emails: List[DirectoryUserEmail] = []
+        for email in attributes.get("emails"):
+            self.emails.push(DirectoryUserEmail(email))
+        self.username: Optional[str] = attributes.get("username")
+        self.state: DirectoryUserState = DirectoryUserState(attributes["state"])
+        self.custom_attributes: JsonDict = attributes.get("custom_attributes", {})
+        self.raw_attributes: JsonDict = attributes.get("raw_attributes", {})
+        self.created_at: str = attributes["created_at"]
+        self.updated_at: str = attributes["updated_at"]
+        self.role: Optional[DirectoryUserRole] = (
```

Comment:
Is this actually optional anymore?

Reply:
I didn't think so, but it's still optional in the API event schema, so will keep as-is until the API is updated.

Reply:
It's the API resource that led me to ask this question. The role here always exists when returning a directory user: https://github.com/workos/workos/blob/e95130bf78e01b202c7e7417adf36135cfc5f033/packages/api/src/directory-users/directory-users.controller.ts#L340

Reply:
It's optional! We don't return it on non-scim/Google users

Reply:
https://github.com/workos/workos/blob/main/packages/api-services/src/directory-users/utils/serialize-directory-user.ts#L79

Reply:
@ameesha but in the code I referenced `role` *always* exists because `getRoleFromUserRoleId` returns `Promise<Role>`.

I  see that the serializer accepts `role?`, but in this code path I don't see how it can ever actually happen.

Reply:
yeah the role is always grabbed but in the serializer (line i sent), we only use it for scim/google dirs

Reply:
nothing cares about whether the role is relevant in the directory type or not but the serializer. it's the only* thing that checks that logic (cause we create and set roles across the board). 

* there are other checks for directory type so we can end recalculations early but if we removed it, nothing would really change